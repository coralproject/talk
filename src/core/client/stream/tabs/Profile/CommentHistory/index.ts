export {
  default,
  default as CommentHistoryContainer,
} from "./CommentHistoryContainer";
