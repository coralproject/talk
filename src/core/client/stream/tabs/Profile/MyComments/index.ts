export { default, default as MyCommentsContainer } from "./MyCommentsContainer";
