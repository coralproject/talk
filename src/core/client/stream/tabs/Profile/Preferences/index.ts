export {
  default,
  default as PreferencesContainer,
} from "./PreferencesContainer";
