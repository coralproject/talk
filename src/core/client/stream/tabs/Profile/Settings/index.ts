export {
  default,
  default as AccountSettingsContainer,
} from "./AccountSettingsContainer";
