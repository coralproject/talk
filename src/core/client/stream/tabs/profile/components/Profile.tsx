import * as React from "react";
import { StatelessComponent } from "react";
import { PropTypesOf } from "talk-framework/types";
import UserBoxContainer from "talk-stream/containers/UserBoxContainer";
import { HorizontalGutter } from "talk-ui/components";
import CommentHistoryContainer from "../containers/CommentHistoryContainer";

export interface ProfileProps {
  story: PropTypesOf<typeof CommentHistoryContainer>["story"];
  me: PropTypesOf<typeof UserBoxContainer>["me"] &
    PropTypesOf<typeof CommentHistoryContainer>["me"];
}

const Profile: StatelessComponent<ProfileProps> = props => {
  return (
    <HorizontalGutter size="double">
      <UserBoxContainer me={props.me} />
      <CommentHistoryContainer me={props.me} story={props.story} />
    </HorizontalGutter>
  );
};

export default Profile;
