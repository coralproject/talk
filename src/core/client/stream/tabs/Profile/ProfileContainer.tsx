import React from "react";
import { graphql } from "react-relay";

import { withFragmentContainer } from "coral-framework/lib/relay";
import { ProfileContainer_settings as SettingsData } from "coral-stream/__generated__/ProfileContainer_settings.graphql";
import { ProfileContainer_story as StoryData } from "coral-stream/__generated__/ProfileContainer_story.graphql";
import { ProfileContainer_viewer as ViewerData } from "coral-stream/__generated__/ProfileContainer_viewer.graphql";

import Profile from "./Profile";

interface ProfileContainerProps {
  viewer: ViewerData;
  settings: SettingsData;
  story: StoryData;
}

export class ProfileContainer extends React.Component<ProfileContainerProps> {
  public render() {
    return (
      <Profile
        viewer={this.props.viewer}
        story={this.props.story}
        settings={this.props.settings}
      />
    );
  }
}
const enhanced = withFragmentContainer<ProfileContainerProps>({
  story: graphql`
    fragment ProfileContainer_story on Story {
      ...CommentHistoryContainer_story
    }
  `,
  viewer: graphql`
    fragment ProfileContainer_viewer on User {
      ...UserBoxContainer_viewer
      ...CommentHistoryContainer_viewer
      ...SettingsContainer_viewer
      ...ChangeUsernameContainer_viewer
      ...ChangeEmailContainer_viewer
      ...DeletionRequestCalloutContainer_viewer
    }
  `,
  settings: graphql`
    fragment ProfileContainer_settings on Settings {
      ...UserBoxContainer_settings
      ...SettingsContainer_settings
    }
  `,
})(ProfileContainer);

export default enhanced;
