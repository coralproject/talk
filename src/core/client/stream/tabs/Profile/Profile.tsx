import * as React from "react";
import { FunctionComponent } from "react";

import { PropTypesOf } from "coral-framework/types";
import UserBoxContainer from "coral-stream/common/UserBox";
import { HorizontalGutter } from "coral-ui/components";

import CommentHistoryContainer from "./CommentHistory";

export interface ProfileProps {
  story: PropTypesOf<typeof CommentHistoryContainer>["story"];
  viewer: PropTypesOf<typeof UserBoxContainer>["viewer"] &
    PropTypesOf<typeof CommentHistoryContainer>["viewer"];
  settings: PropTypesOf<typeof UserBoxContainer>["settings"];
}

const Profile: FunctionComponent<ProfileProps> = props => {
  return (
    <HorizontalGutter size="double">
      <UserBoxContainer viewer={props.viewer} settings={props.settings} />
      <CommentHistoryContainer viewer={props.viewer} story={props.story} />
    </HorizontalGutter>
  );
};

export default Profile;
