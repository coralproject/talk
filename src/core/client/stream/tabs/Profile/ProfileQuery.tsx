import { Localized } from "@fluent/react/compat";
import { once } from "lodash";
import React, { FunctionComponent, Suspense } from "react";
import { graphql } from "react-relay";

import { polyfillCSSVars } from "coral-framework/helpers";
import {
  QueryRenderData,
  QueryRenderer,
  withLocalStateContainer,
} from "coral-framework/lib/relay";
import useHandleIncompleteAccount from "coral-stream/common/useHandleIncompleteAccount";
import { CallOut, Delay, Spinner } from "coral-ui/components/v2";

import { ProfileQuery as QueryTypes } from "coral-stream/__generated__/ProfileQuery.graphql";
import { ProfileQueryLocal as Local } from "coral-stream/__generated__/ProfileQueryLocal.graphql";

const loadProfileContainer = () =>
  import("./ProfileContainer" /* webpackChunkName: "profile" */).then((x) => {
    // New css is loaded, take care of polyfilling those css vars for IE11.
    void polyfillCSSVars();
    return x;
  });
// (cvle) For some reason without `setTimeout` this request will block other requests.
const preloadProfileContainer = once(() => setTimeout(loadProfileContainer, 0));

const LazyProfileContainer = React.lazy(loadProfileContainer);

interface Props {
  local: Local;
}

export const render = ({ error, props }: QueryRenderData<QueryTypes>) => {
  if (error) {
    return (
      <CallOut color="error" fullWidth>
        {error.message}
      </CallOut>
    );
  }

  // TODO: use official React API once it has one :-)
  preloadProfileContainer();

  if (props) {
    if (!props.viewer) {
      return (
        <Localized id="profile-profileQuery-errorLoadingProfile">
          <CallOut color="error" fullWidth>
            Error loading profile
          </CallOut>
        </Localized>
      );
    }
    if (!props.story) {
      return (
        <Localized id="profile-profileQuery-storyNotFound">
          <CallOut>Story not found</CallOut>
        </Localized>
      );
    }
    return (
      <Suspense fallback={<Spinner />}>
        <LazyProfileContainer
          viewer={props.viewer}
          story={props.story}
          settings={props.settings}
        />
      </Suspense>
    );
  }

  return (
    <Delay>
      <Spinner />
    </Delay>
  );
};

const ProfileQuery: FunctionComponent<Props> = ({
  local: { storyID, storyURL },
}) => {
  const handleIncompleteAccount = useHandleIncompleteAccount();
  return (
    <QueryRenderer<QueryTypes>
      query={graphql`
        query ProfileQuery($storyID: ID, $storyURL: String) {
          story: stream(id: $storyID, url: $storyURL) {
            ...ProfileContainer_story
          }
          viewer {
            ...ProfileContainer_viewer
          }
          settings {
            ...ProfileContainer_settings
          }
        }
      `}
      variables={{
        storyID,
        storyURL,
      }}
      render={(data) => {
        if (handleIncompleteAccount(data)) {
          return null;
        }
        return render(data);
      }}
    />
  );
};

const enhanced = withLocalStateContainer(
  graphql`
    fragment ProfileQueryLocal on Local {
      storyID
      storyURL
    }
  `
)(ProfileQuery);

export default enhanced;
