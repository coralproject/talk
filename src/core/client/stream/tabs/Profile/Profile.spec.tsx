import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import { removeFragmentRefs } from "coral-framework/testHelpers";
import { PropTypesOf } from "coral-framework/types";

import Profile from "./Profile";

const ProfileN = removeFragmentRefs(Profile);

it("renders correctly", () => {
  const props: PropTypesOf<typeof ProfileN> = {
    story: {},
    viewer: {},
    settings: {},
  };
  const renderer = createRenderer();
  renderer.render(<ProfileN {...props} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
