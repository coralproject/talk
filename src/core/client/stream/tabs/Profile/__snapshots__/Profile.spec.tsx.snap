// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(forwardRef)
  size="double"
>
  <withContext(createMutationContainer(withContext(createMutationContainer(withContext(createMutationContainer(withContext(withLocalStateContainer(Relay(UserBoxContainer)))))))))
    settings={Object {}}
    viewer={Object {}}
  />
  <Relay(CommentHistoryContainer)
    story={Object {}}
    viewer={Object {}}
  />
</ForwardRef(forwardRef)>
`;
